eknrenderer/eknr-renderer.c

```diff
@@ -0,0 +1,1306 @@
+}
+
+/**
+ * eknr_renderer_render_mustache_document:
+ * @renderer: An #EknrRenderer
+ * @tmpl_text: The template to render
+ * @variables: The variables and sections to use when rendering.
+ * @error: A #GError
+ *
+ * Use mustache_c to render a document. The provided @variables variant
+ * is used as substitutions. The variant should be of type
+ * 'a{sv}' and each child node should be either 's' or a variable subsitution
+ * 'as' for a section substitution.
+ *
+ * Returns: (transfer full): The renderered document on success, %NULL on error.
+ */
+char *
+eknr_renderer_render_mustache_document (G_GNUC_UNUSED EknrRenderer  *renderer,
+                                        const char                  *tmpl_text,
+                                        GVariant                    *variables,
+                                        GError                     **error)
+{
+  g_autoptr(GVariantDict) variables_dict = g_variant_dict_new (variables);
+  g_autoptr(RendererMustacheData) data = renderer_mustache_data_new (variables_dict,
+                                                                     error,
+                                                                     tmpl_text);
+  g_autoptr(mustache_template_t) tmpl = mustache_compile (&_renderer_mustache_data_vfuncs,
+                                                          data);
+
+  if (!tmpl)
+    return NULL;
+
+  return _renderer_render_mustache_document_internal (tmpl, variables, error);
+}
+
+/**
+ * eknr_mustache_template_compiles:
+ * @tmpl_text: The template to render
+ * @error: A #GError
+ *
+ * Attempt to compile a mustache template.
+ *
+ * Returns: The %TRUE on success, %FALSE on error.
+ */
+gboolean
+eknr_mustache_template_compiles (const char   *tmpl_text,
+                                 GError      **error)
+{
+  g_autoptr(RendererMustacheData) data = renderer_mustache_data_new (NULL,
+                                                                     error,
+                                                                     tmpl_text);
+  g_autoptr(mustache_template_t) tmpl = mustache_compile (&_renderer_mustache_data_vfuncs,
+                                                          (char *) data);
+
+  if (!tmpl)
+    return FALSE;
+
+  return TRUE;
+}
+
+static char *
+format_a_href_link (const char   *uri,
+                    const char   *text,
+                    GError      **error)
+{
+  g_autofree char *escaped = eknr_escape_html (text, error);
+
+  if (escaped == NULL)
+    return NULL;
+
+  return g_strdup_printf ("<a class=\"eos-show-link\" href=\"%s\">%s</a>",
+                          uri,
+                          escaped);
+}
+
+static char *
+format_license_link (const char   *license,
+                     GError      **error)
+{
+  g_autofree char *escaped = g_uri_escape_string (license, NULL, TRUE);
+  g_autofree char *licence_link = g_strdup_printf ("license://%s", escaped);
```

Comment:
Be consistent with license / licence (I'd prefer the US spelling since that's already used in the URI scheme)

Reply:
Agreed